src/AppKit/NSView.cs

```diff
@@ -0,0 +1,55 @@
+//
+// Copyright 2019 Microsoft Corp.
+//
+
+using System;
+using System.Runtime.InteropServices;
+
+using Foundation;
+using ObjCRuntime;
+
+namespace AppKit {
+
+	public partial class NSView {
+
+		delegate nint view_compare_func (IntPtr view1, IntPtr view2, IntPtr context);
+		static view_compare_func view_comparer = view_compare;
+
+		class SortData
+		{
+			public Exception Exception;
+			public Func<NSView, NSView, NSComparisonResult> Comparer;
+		}
+
+		[MonoPInvokeCallback (typeof (view_compare_func))]
+		static nint view_compare (IntPtr view1, IntPtr view2, IntPtr context)
+		{
+			var data = (SortData) GCHandle.FromIntPtr (context).Target;
+			try {
+				var a = (NSView) Runtime.GetNSObject (view1);
+				var b = (NSView) Runtime.GetNSObject (view2);
+				return (nint) (long) data.Comparer (a, b);
+			} catch (Exception e) {
+				data.Exception = e;
+				return (nint) (long) NSComparisonResult.Same;
+			}
+		}
+
+		public void SortSubviews (Func<NSView, NSView, NSComparisonResult> comparer)
```

Comment:
we generally define a custom delegate when there's more than one generic argument - but I think this one is self documenting.

Reply:
It might be nice to have a `Comparison<T>` override though.

Reply:
We can't propagate `T` to the P/Invoke callback (`view_compare`), so a `Comparison<T>` overload wouldn't be trivial to implement (it would probably need to use reflection to call the managed comparison function).